src/couch_mrview/src/couch_mrview.erl

```diff
@@ -532,35 +532,61 @@ map_fold({{Key, Id}, Val}, _Offset, Acc) ->
         last_go = Go
     }};
 map_fold(#doc{id = <<"_local/", _/binary>>} = Doc, _Offset, #mracc{} = Acc) ->
-    #mracc{
-        limit = Limit,
-        callback = Callback,
-        user_acc = UAcc0,
-        args = Args
-    } = Acc,
-    #doc{
-        id = DocId,
-        revs = {Pos, [RevId | _]}
-    } = Doc,
-    Rev = {Pos, RevId},
-    Row =
-        [
-            {id, DocId},
-            {key, DocId},
-            {value, {[{rev, couch_doc:rev_to_str(Rev)}]}}
-        ] ++
-            if
-                not Args#mrargs.include_docs -> [];
-                true -> [{doc, couch_doc:to_json_obj(Doc, Args#mrargs.doc_options)}]
-            end,
-    {Go, UAcc1} = Callback({row, Row}, UAcc0),
-    {Go, Acc#mracc{
-        limit = Limit - 1,
-        reduce_fun = undefined,
-        doc_info = undefined,
-        user_acc = UAcc1,
-        last_go = Go
-    }}.
+    IncludeSys = couch_util:get_value(include_system, Acc#mracc.args#mrargs.extra),
+
+    case should_skip_local_document(IncludeSys, Doc) of
+        true ->
+            {ok, Acc};
+        _ ->
+            #mracc{
+                limit = Limit,
+                callback = Callback,
+                user_acc = UAcc0,
+                args = Args
+            } = Acc,
+            #doc{
+                id = DocId,
+                revs = {Pos, [RevId | _]}
+            } = Doc,
+            Rev = {Pos, RevId},
+            Row =
+                [
+                    {id, DocId},
+                    {key, DocId},
+                    {value, {[{rev, couch_doc:rev_to_str(Rev)}]}}
+                ] ++
+                    if
+                        not Args#mrargs.include_docs -> [];
+                        true -> [{doc, couch_doc:to_json_obj(Doc, Args#mrargs.doc_options)}]
+                    end,
+            {Go, UAcc1} = Callback({row, Row}, UAcc0),
+            {Go, Acc#mracc{
+                limit = Limit - 1,
+                reduce_fun = undefined,
+                doc_info = undefined,
+                user_acc = UAcc1,
+                last_go = Go
+            }}
+    end.
+
+should_skip_local_document(_IncludeSys = false, #doc{
```

Comment:
I don't think we need to be quite so picky on the contents, skipping based on the id is sufficient.

Reply:
It's plausible a user would have a "_local/purge-1" document of some sort. `"shard-sync-..."` probably a lot less likely that was the reason for the extra fields check.

Reply:
hm, ok.